src/main/java/org/example/jvspringbootfirstbook/service/BookServiceImpl.java

```diff
@@ -1,25 +1,39 @@
 package org.example.jvspringbootfirstbook.service;
 
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.jvspringbootfirstbook.dto.BookDto;
+import org.example.jvspringbootfirstbook.dto.CreateBookRequestDto;
+import org.example.jvspringbootfirstbook.exception.EntityNotFoundException;
+import org.example.jvspringbootfirstbook.mapper.BooksMapper;
 import org.example.jvspringbootfirstbook.model.Book;
 import org.example.jvspringbootfirstbook.repository.BookRepository;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BooksMapper bookMapping;
 
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapping.toModel(createBookRequestDto);
+        return bookMapping.toBookDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapping::toBookDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book with the id: " + id)
+        );
+        return bookMapping.toBookDto(book);
     }
```

Comment:
A bit more readable approach (example):

```suggestion
    public BookDto findById(Long id) {
        return bookMapper.toDto(
                bookRepository.findById(id).orElseThrow(
                        () -> new EntityNotFoundException("Can't find book with id: " + id)));
    }
```